File: renders/habitats/LandHabitat.cpp
/* NIM/Nama  : 13515AAA / AAA       */
/* File      : Habitat.cpp              */

#include "LandHabitat.h"

using namespace std;

LandHabitat::LandHabitat() : id('X'), color(WHITE)
{
}
LandHabitat::LandHabitat(const LandHabitat&) : id('X'), color(WHITE)
{
}
LandHabitat::~LandHabitat()
{
}
LandHabitat& LandHabitat::operator=(const LandHabitat&)
{
}
LandHabitat* LandHabitat::clone() const
{
	return new LandHabitat(*this);
}

File: LandHabitat.java
/**
 * @author NIM/Nama : 13515078 / Veren Iliana Kurniadi
 * File : LandHabitat.java
 */

import java.util.*;
import java.lang.*;

public class LandHabitat extends Habitat{
   /**
   * Constructor
   */
  public LandHabitat(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char render(){
    return 'L';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String renderWithColor(){
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}
